core/qthelper.h

```diff
@@ -29,6 +30,18 @@ class QImage;
 #define SKIP_EMPTY QString::SkipEmptyParts
 #endif
 
+// this is annoying Qt5 / Qt6 incompatibility where we can't compare against string literals anymore
```

Comment:
Yeah, we couldn't do much if we needed to remove QStringRef.

But you should be able to write:
```c++
    r.name().compare(QLatin1String(cs))
```

BTW, do note that one returns a memcmp-like result, with 0 being "equal", whereas the == operator returns a boolean, so 1 (true) is equal.

Reply:
I think I correctly adjusted the logic everywhere, but I'll go through that one more time very carefully just in case...

Reply:
brain not working... I now realize what you meant. will be fixed in the next force push

Reply:
Having wasted quite some time with disentangling the include hell, I'm not happy that now every user of `qthelper` slurps in the obscure `QXmlStreamReader` header.

Moreover, the helper function seems ridiculously specific.

Can't we do something like
```
#include <QString>
// Make QStringView and QStringRef behave the same with respect to
// comparison with C-strings.
// Needed, because the Qt v6 API changed from the latter to the former.
inline bool operator==(QStringView &v, const char *s)
{
        return v.compare(QString::fromUtf8(s)) == 0;
}
```
and just keep the rest?

Of course, that would be back to conditional compilation, because QStringView does not exist pre-Qt 5.10 or somesuch. 

To avoid conditional compilation, we could define a much more general template, such as:
```
#include <QString>
template <typename StringView>
bool isSameString(StringView v, const char *s)
{
        return v.compare(QString::fromUtf8(s)) == 0;
}
```
and then use it like so
```
        if (isSameString(gpxReader.name(), "time") {
                ...
        }
```
This should work equally for QStringView and QStringRef.

Reply:
running way behind - had a lovely Sunday morning far away from computers.

I will simply add a separate header file for this that is only included in the couple of places where we do this. Putting this in `qthelper.h` was laziness on my part.